botorch_community/models/gp_regression_multisource.py

```diff
@@ -0,0 +1,401 @@
+        ]
+
+        # Create the training set for the AGP selecting all
+        # the observations from the high fidelity source
+        # and the reliable observations from the other sources
+        reliable_idxs = [
+            _get_reliable_observations(
+                self.models[0], self.models[s], train_X[s][:, :-1], m
+            )
+            for s in sources[1:]
+        ]
+        train_X = torch.cat(
+            [
+                train_X[s] if s == 0 else train_X[s][reliable_idxs[s - 1]]
+                for s in sources
+            ]
+        )[:, :-1]
+        train_Y = torch.cat(
+            [
+                train_Y[s] if s == 0 else train_Y[s][reliable_idxs[s - 1]]
+                for s in sources
+            ]
+        )
+
+        super().__init__(
+            train_X,
+            train_Y,
+            train_Yvar,
+            likelihood,
+            covar_module,
+            mean_module,
+            outcome_transform,
+            input_transform,
+        )
+
+    def _init_fit_gp(
+        self,
+        train_X: Tensor,
+        train_Y: Tensor,
+        likelihood: Optional[Likelihood] = None,
+        covar_module: Optional[Module] = None,
+        mean_module: Optional[Mean] = None,
+        outcome_transform: Optional[OutcomeTransform] = None,
+        input_transform: Optional[InputTransform] = None,
+    ) -> SingleTaskGP:
+        r"""Initialize and fit a Single Task GP model.
+
+        Args:
+            train_X: A `batch_shape x n x d` tensor of training features.
+            train_Y: A `batch_shape x n x m` tensor of training observations.
+            likelihood: A likelihood. If omitted, use a standard
+                GaussianLikelihood with inferred noise level.
+            covar_module: The module computing the covariance (Kernel) matrix.
+                If omitted, use a `MaternKernel`.
+            mean_module: The mean function to be used. If omitted, use a
+                `ConstantMean`.
+            outcome_transform: An outcome transform that is applied to the
+                training data during instantiation and to the posterior during
+                inference (that is, the `Posterior` obtained by calling
+                `.posterior` on the model will be on the original scale).
+            input_transform: An input transform that is applied in the model's
+                forward pass.
+
+        Returns:
+            The fitted Single Task GP and its Marginal Log Likelihood.
+        """
+        gp = SingleTaskGP(
+            train_X,
+            train_Y,
+            likelihood=likelihood,
+            covar_module=covar_module,
+            mean_module=mean_module,
+            outcome_transform=outcome_transform,
+            input_transform=input_transform,
+        )
+        mll = ExactMarginalLogLikelihood(gp.likelihood, gp)
+        fit_gpytorch_mll(mll)
+        return gp
+
+
+class FixedNoiseAugmentedGP(FixedNoiseGP):
```

Comment:
FWIW we have deprecated `FixedNoiseGP` and incorporated its functionality into `SingleTaskGP` by allowing to pass `train_Yvar` also to `SingleTaskGP` (optionally). Would be great to do the same setup here so that the implementations are analogous.

Reply:
I incorporated `FixedNoiseAugmentedAGP` in `SingleTaskAugmentedGP` (and consequently the unit tests).